feat: inbox preference grouping and shadow root detection (#846)
novuhq/docs: content/docs/platform/inbox/react/components/preferences.mdx

  Note: Grouping preference is supported in client-side SDKs starting from version 3.4.0.
</Callout>

To implement preference grouping, use the `preferenceGroups` prop on the `Inbox` component. Each group requires:

- `name`: The text displayed for the group
- `filter`: Criteria for which preferences appear in the group (object or function)

Here are the key ways to filter preferences:

#### Filter by tags
Use this approach to group preferences that share specific tags.

   ```tsx
   const preferenceGroups = [
  {
    name: 'General',
    filter: {
      tags: ['account'],
    },
  },
];
   ```

#### Filter by tags and workflow IDs:
Combine tags and workflow IDs to define a more targeted group of preferences.

   ```tsx
   const preferenceGroups = [
  {
    name: 'Marketing',
    filter: {
      tags: ['marketing'],
      workflowIds: ['workflow-id'],
    },
  },
];

   ```

#### Include all workflows
Include every available workflow, regardless of its tag or grouping, by returning the full preferences list.

   ```tsx
   const preferenceGroups = [
  {
    name: 'All Workflows',
    filter: ({ preferences }) => preferences,
  },
];
  ```

#### Custom filtering logic

>    You can implement custom filtering logic using a function that receives the preferences array (excluding global preferences) and returns filtered preferences based on your criteria.

Reviewer: ```suggestion
You can implement custom filtering logic using a function that receives the preferences array (excluding global preferences) and returns filtered preferences based on your criteria.
```

  display: none;
}
```

### Grouping Preferences

> Preference Grouping allows you to organize extensive lists of workflows into well-organized, meaningful groups. This ensures that your end-users get a better experience when managing their preferences.

Reviewer 2: ```suggestion
Preference grouping lets you organize extensive lists of workflows into well-organized, meaningful groups. This ensures that your end-users get a better experience when managing their preferences.
```

To implement preference grouping, use the `preferenceGroups` prop on the `Inbox` component. Each group requires:

- `name`: The text displayed for the group
- `filter`: Criteria for which preferences appear in the group (object or function)

Here are the key ways to filter preferences:

> 1. **Filter by tags**:

Reviewer 2: ```suggestion
- **Filter by tags**:
```
Numbers are only used for sequential steps

   ```tsx
   const groupConfig = { name: 'General', filter: { tags: ['account'] } };
   ```

> 2. **Filter by tags and workflow IDs**:

Reviewer 2: ```suggestion
- **Filter by tags and workflow IDs**:
```

   ```tsx
   const groupConfig = { name: 'Marketing', filter: { tags: ['marketing'], workflowIds: ['workflow-id'] } };
   ```

> 3. **Include all workflows** (even those in other groups):

Reviewer 2: ```suggestion
- **Include all workflows** (even those in other groups):
```

   ```tsx
   const groupConfig = { name: 'All Workflows', filter: ({ preferences }) => preferences };
   ```

> 4. **Custom filtering logic**:

Reviewer 2: ```suggestion
- **Custom filtering logic**:
```

   You can implement custom filtering logic using a function that receives the preferences array (excluding global preferences) and returns filtered preferences based on your criteria.

> Example implementation:

Reviewer 2: We need to include a screenshot of how this appears in the UI

  display: none;
}
```

### Grouping preferences

Preference grouping lets you organize extensive lists of workflows into well-organized, meaningful groups. This ensures that your end-users get a better experience when managing their preferences.
 
<Callout type="info">
  Note: Grouping preference is supported in client-side SDKs starting from version 3.4.0.
</Callout>

To implement preference grouping, use the `preferenceGroups` prop on the `Inbox` component. Each group requires:

- `name`: The text displayed for the group
- `filter`: Criteria for which preferences appear in the group (object or function)

Here are the key ways to filter preferences:

- **Filter by tags**:
   ```tsx
   const groupConfig = { name: 'General', filter: { tags: ['account'] } };
   ```

- **Filter by tags and workflow IDs**:
   ```tsx
   const groupConfig = { name: 'Marketing', filter: { tags: ['marketing'], workflowIds: ['workflow-id'] } };
   ```

- **Include all workflows** (even those in other groups):
   ```tsx
   const groupConfig = { name: 'All Workflows', filter: ({ preferences }) => preferences };

>    ```

Reviewer 3: @Aviatorscode2 let's maybe make those snippets not a single line? But prettier them? Because right now they are not the best to read. 

Also, can we maybe consider a different display for them and not in bullets (not sure I like how it looks like(
